internal/service/ec2/ec2_eip.go

```diff
@@ -278,6 +278,8 @@ func resourceEIPRead(ctx context.Context, d *schema.ResourceData, meta any) diag
 		d.Set("ptr_record", strings.TrimSuffix(aws.ToString(addressAttr.PtrRecord), "."))
 	case tfresource.NotFound(err):
 		d.Set("ptr_record", nil)
+	case tfawserr.ErrMessageContains(err, "InvalidAction", "not valid for this web service"):
+		// Effectively ignoring the error as we handle errors from GovCloud/Secret Regions
```

Comment:
we should set `ptr_record` to `nil`, similar to when `NotFound(err)` is checked.